Implementation for triSYCL issue #120

yu810226 opened the pull request:
Implementation for triSYCL issue #120

yu810226 committed:
Set metadata for reqd_work_group_size (1,1,1)

Set metadata for reqd_work_group_size (1,1,1) while compiling for
FPGA.
Add -reqd-workgroup-size-fpga to control wheter adding this metadata or not.

keryell commented on lib/SYCL/inSPIRation.cpp:
```diff
@@ -41,6 +41,8 @@ using namespace llvm;
 /// Displayed with -stats
 STATISTIC(SYCLKernelProcessed, "Number of SYCL kernel functions processed");
 
+cl::opt<bool> ReqdWorkGroupSizeFPGA("reqd-workgroup-size-fpga", cl::desc("set reqd_work_group_size to be 1-1-1"));
```
Would be better with a more modern syntax with `{ }` everywhere instead of `( )`

keryell replied:
It is not clear to me why there is `FPGA` or `-fpga` in the name.

keryell replied:
Should the option would be named `sycl-reqd-workgroup-size-1` instead?

yu810226 replied:
Would be better with a more modern syntax with { } everywhere instead of ( ).
Not sure about this. I checked *cl::opt<bool>* and *cl::desc* using in LLVM other passes are all with () instead of {}. Do you think keeping () is a good idea.


Should the option would be named *sycl-reqd-workgroup-size-1* instead?
Yes, I think naming is better.

Thanks! :-)

keryell reviewed (changes requested):
That looks useful!
Some details could be improved and the loop-idiom commit should not be in this PR.
Can you remove it and force-push again after the other changes?
Thanks.

yu810226 committed:
Change opt flag name for reqd-workgroup-size-fpga

Change ReqdWorkGroupSizeFPGA to ReqdWorkGroupSizeOne
Change reqd-workgroup-size-fpga to reqd-workgroup-size-1

keryell commented:
Cool.
